samtranslator/swagger/swagger.py

```diff
@@ -612,14 +612,16 @@ def set_path_default_authorizer(  # noqa: too-many-branches
                 if "AWS_IAM" in method_definition["security"][0]:
                     self.add_awsiam_security_definition()
 
-    def set_path_default_apikey_required(self, path: str) -> None:
+    def set_path_default_apikey_required(self, path: str, required_options_api_key: bool = True) -> None:
```

Comment:
is `required_options_api_key` optional with True as the default value?

Reply:
Yeah. Default value should be True and it is optional.

Reply:
Should be Optional[bool] = True then i think

Reply:
Let's take a step back to Python without types. `required_options_api_key=True` means if `required_options_api_key` not passed to the function, set it to `True`. Compare it to `required_options_api_key=None`. It'd mean, if not passed to the function, set it to `None`. Now, `Optional` is equivalent to `Union[some_type, None]`. So `Optional[bool]` means `Union[bool, None]` and it actually means 3 possible values `True`, `False`, `None`. While `required_options_api_key: bool = True` still means only 2 possible values `True` and `False` with `True` if the variable is not passed to the function (it actually allows not to pass a variable). Connor is right.
P.S I hate typing in Python (and Typescript too)

Reply:
That was a really good explanation. I'll change it back.

Reply:
yep, thanks for the explanation slava